lib/cn/intern_builder.c

```diff
@@ -56,20 +66,30 @@ struct intern_level {
     struct intern_node     *node_head;
     struct intern_node     *node_curr;
     unsigned char          *sbuf;
-    uint                    sbuf_sz;
-    uint                    sbuf_used;
+    size_t                  sbuf_sz;
+    size_t                  sbuf_used;
     struct intern_level    *parent;
     struct intern_builder  *ibldr;
 };
 
-/* Max sizes of objects embedded into struct intern_builder.
+/* Max sizes of objects embedded into struct intern_builder.  For release
+ * builds, the initial embedded buffer size is large enough that calls to
+ * realloc() should rare for keys of reasonable length (i.e. < 64 bytes).
+ * We make it smaller for non-release builds in order to better exercise
+ * the realloc() path.
  */
+#ifdef HSE_BUILD_RELEASE
+#define IB_ESBUFSZ_KMC      (1u << 16) /* 64K ibldr kmem cache size */
+#else
+#define IB_ESBUFSZ_KMC      (1u << 13) /* 8K ibldr kmem cache size */
+#endif
+
 #define IB_ENODEV_MAX       NELEM(((struct intern_builder *)0)->nodev)
 #define IB_ELEVELV_MAX      NELEM(((struct intern_builder *)0)->levelv)
-#define IB_ESBUFSZ_MAX                                                  \
-    roundup(                                                            \
-        ((8192 - sizeof(struct intern_builder) - 16) / IB_ELEVELV_MAX), \
-        alignof(struct intern_key))
+#define IB_ESBUFSZ_MAX                                                               \
+    roundup(                                                                         \
+        ((IB_ESBUFSZ_KMC - sizeof(struct intern_builder) - 16) / IB_ELEVELV_MAX),    \
+          alignof(struct intern_key))
```

Comment:
This didn't prevent LTO from running, but the LTO runtime failures got me scrutinizing this code, the new confguration wastes less space as we no longer need to support a 5-level tree, and no longer perform lots of reallocs().

Reply:
This is the WBTree, not the CN tree.  What changed to reduce the max levels on the WBTree?

Reply:
Yeah, I understand that now.  But this is just a cache, it doesn't limit anything.  So it's optimized now for keys < 64-bytes, still trying to find the edge.  It may be I increase it to 3, but for almost everything we test it's a waste of cache that we can better use to accomodate common use cases.